### DISCUSSION TITLE
Lesson 9 - function testCheckUpkeepReturnFalseIfRaffleIsntOpen

### DISCUSSION THREAD

**Discussion by Merulez** [Cyfrin/foundry-full-course-cu#1110] [Q&A]
function testFullRandomWordsPicksAWinnerResetAndSendMoney()
        public
        raffleEnteredAndTimePassed
    {
        uint256 additionalEntrants = 5;
        uint256 startingIndex = 1;
        for (
            uint256 i = startingIndex;
            i < startingIndex + additionalEntrants;
            ++i
        ) {
            address player = address(uint160(i));
            hoax(player, 1 ether);
            raffle.enterRaffle{value: entranceFee}();
        }
        uint256 prize = entranceFee * (additionalEntrants + startingIndex);
        uint256 winnerBalance = raffle.getRecentWinner().balance;

        vm.recordLogs();
        raffle.perfromUpKeep("");
        Vm.Log[] memory entries = vm.getRecordedLogs();
        bytes32 requestId = entries[1].topics[1];

        uint256 previousTimeStamp = raffle.getLastTimeStamp();
        // pretend to be a chainlink vrf to get a random number and pick winner
        VRFCoordinatorV2Mock(vrfCoordinator).fulfillRandomWords(
            uint256(requestId),
            address(raffle)
        );
        // Assert
        // assert(uint256(raffle.getRaffleState()) == 0);
        // assert(raffle.getRecentWinner() != address(0));
        // assert(raffle.getLengthOfPlayers() == 0);
        // assert(previousTimeStamp < raffle.getLastTimeStamp());
        console.log(raffle.getRecentWinner().balance);
        console.log(prize);
        console.log(STARTING_USER_BALANCE + prize - entranceFee);
        assert(winnerBalance == STARTING_USER_BALANCE + prize);
    }
1050000000000000000
  60000000000000000
10050000000000000000    // This is what i get from the logs i dont know why

**Comment by MarcoW3** [comment]
You have to set `hoax(player, 1 ether);` as `hoax(player, STARTING_USER_BALANCE);`.
